source/extensions/filters/common/expr/context.cc

```diff
@@ -29,119 +29,450 @@ convertHeaderEntry(Protobuf::Arena& arena,
                    Http::HeaderUtility::GetAllOfHeaderAsStringResult&& result) {
   if (!result.result().has_value()) {
     return {};
-  } else if (!result.backingString().empty()) {
+  }
+  if (!result.backingString().empty()) {
     return CelValue::CreateString(
         Protobuf::Arena::Create<std::string>(&arena, result.backingString()));
-  } else {
-    return CelValue::CreateStringView(result.result().value());
   }
+  return CelValue::CreateStringView(result.result().value());
 }
 
-namespace {
+// SSL Extractors implementation
+const SslExtractorsValues& SslExtractorsValues::get() {
+  CONSTRUCT_ON_FIRST_USE(
+      SslExtractorsValues,
+      absl::flat_hash_map<absl::string_view, SslExtractor>{
+          {TLSVersion,
+           [](const Ssl::ConnectionInfo& info) {
+             return CelValue::CreateString(&info.tlsVersion());
+           }},
+          {SubjectLocalCertificate,
+           [](const Ssl::ConnectionInfo& info) {
+             return CelValue::CreateString(&info.subjectLocalCertificate());
+           }},
+          {SubjectPeerCertificate,
+           [](const Ssl::ConnectionInfo& info) {
+             return CelValue::CreateString(&info.subjectPeerCertificate());
+           }},
+          {URISanLocalCertificate,
+           [](const Ssl::ConnectionInfo& info) {
+             return !info.uriSanLocalCertificate().empty()
+                        ? CelValue::CreateString(&info.uriSanLocalCertificate()[0])
+                        : CelValue::CreateNull();
+           }},
+          {URISanPeerCertificate,
+           [](const Ssl::ConnectionInfo& info) {
+             return !info.uriSanPeerCertificate().empty()
+                        ? CelValue::CreateString(&info.uriSanPeerCertificate()[0])
+                        : CelValue::CreateNull();
+           }},
+          {DNSSanLocalCertificate,
+           [](const Ssl::ConnectionInfo& info) {
+             return !info.dnsSansLocalCertificate().empty()
+                        ? CelValue::CreateString(&info.dnsSansLocalCertificate()[0])
+                        : CelValue::CreateNull();
+           }},
+          {DNSSanPeerCertificate,
+           [](const Ssl::ConnectionInfo& info) {
+             return !info.dnsSansPeerCertificate().empty()
+                        ? CelValue::CreateString(&info.dnsSansPeerCertificate()[0])
+                        : CelValue::CreateNull();
+           }},
+          {SHA256PeerCertificateDigest, [](const Ssl::ConnectionInfo& info) {
+             return !info.sha256PeerCertificateDigest().empty()
+                        ? CelValue::CreateString(&info.sha256PeerCertificateDigest())
+                        : CelValue::CreateNull();
```

Comment:
This is changing the semantics I believe? Previously `has(connection.dns_san_peer_certificate)` would evaluate to false because the attribute is missing. Now it's evaluating to false because it's present but has the value `null`?

Reply:
I agree. This would return an optional containing a `null` CelValue. I'll switch it to `{}` like before. Apologies for the oversight.